Update to highlight the impact of the global state setting (#57067)
elastic/elasticsearch: docs/reference/snapshot-restore/restore-snapshot.asciidoc

{
  "indices": "index_1,index_2",
  "ignore_unavailable": true,

>   "include_global_state": true,              <1>

Reviewer: I really think we should use `false` in the example here and adjust the callout to match, with the expectation that people will continue to copy-paste this anyway.

```suggestion
  "include_global_state": false,              <1>
```

  "rename_pattern": "index_(.+)",

>   "rename_replacement": "restored_index_$1"

Reviewer: Suggest adding `include_aliases` to the example too:

```suggestion
  "rename_replacement": "restored_index_$1",
  "include_aliases": false
```

}
-----------------------------------
// TEST[continued]

<1> By default, `include_global_state` is `false`, meaning the snapshot's
cluster state is not restored.
+
If `true`, the snapshot's persistent settings, index templates, ingest

> pipelines, and {ilm-init} policies are restored and added to the current

Reviewer: ```suggestion
pipelines, and {ilm-init} policies are restored into the current
```

> cluster. This overrides current settings, templates, pipeline, and policies with

Reviewer: I think "overrides" is still a bit ambiguous. How about "overwrites"?

```suggestion
cluster. This overwrites any existing cluster settings, templates, pipelines and {ilm-init} policies whose names match those in the snapshot.
```

      "repository": "my_backup",
      "uuid": "XuBo4l4ISYiVg0nYUen9zg",
      "state": "SUCCESS",

>       "include_global_state": false,

Reviewer: I think `true` is ok here, this is in an API response.

Reviewer 2: I reverted this with a07e2a06fe4d9f64f56467bd3721b31de1ad6800.
